### DISCUSSION TITLE
DataTable to Excel template

### DISCUSSION THREAD

**Discussion by SirParser** [mini-software/MiniExcel#354] [Q&A]
I'm trying to output DataTable to existing Excel template.
This is the template:
![image](https://user-images.githubusercontent.com/102298858/160849001-80148ec5-bdab-4a32-b75a-a0d09e9df85d.png)

I'm populating a data table in PowerShell and then running this:
`[MiniExcelLibs.MiniExcel]::SaveAsByTemplate($ExcelPath, $template, $DataTable)`

The output file is empty:
![image](https://user-images.githubusercontent.com/102298858/160850052-c0e23b5a-1526-43de-9b9c-1cae5b9dbc35.png)

What am I doing wrong or is it not supported?

**Comment by shps951023** [comment]
@HiTekBot Please use level 2 template string
![image](https://user-images.githubusercontent.com/12729184/161413827-87489d8b-4600-44c5-84e7-e7539db49901.png)

**Reply by SirParser** [reply]
Thanks for the feedback.
Unfortunately I'm still stuck.

Template:
![image](https://user-images.githubusercontent.com/102298858/164551558-edf5e90e-4003-4822-8f23-2cd9668952c2.png)
```
$managers = New-Object System.Data.DataTable
    $managers.Columns.Add("name");
    $managers.Columns.Add("department");
    $managers.Rows.Add("Jack", "HR");
    $managers.Rows.Add("Loan", "IT");

$value = @{title = "FooCompany"; managers = $managers}

[MiniExcelLibs.MiniExcel]::SaveAsByTemplate($FilePath, $Template, $value);
```

**Error:**
_Exception calling "SaveAsByTemplate" with "3" argument(s): "Parameter count mismatch."_

**Reply by SirParser** [reply]
It turns out that MiniExcel doesn't recognize HashTable class, it has to be System.Collections.Generic.Dictionary"[String,Object]"
In PowerShell it would be:
    $Value = New-Object System.Collections.Generic.Dictionary"[String,Object]"
    $Value.Add("title", "FooCompany")
    $Value.Add("managers", $managers)
